#155319173, #155319231, #155183816, #155922706 Add the navigation view - OnHold

muhallan opened the pull request:
#### What does this PR do?
Adds the navigation view to the SerialEntryActivity
#### Description of Task to be completed?
- Ensure the hamburger icon is visible
- Add clickable menu items to the navigation view
- Write tests for the navigation view
#### How should this be manually tested?
Checkout to this branch, run the app, then click hamburger icon, or swipe right to draw the navigation menu and swipe left to close it.
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/155319173
- https://www.pivotaltracker.com/story/show/155319231
- https://www.pivotaltracker.com/story/show/155183816
#### Screenshots (if appropriate)
Hamburger Icon
![image](https://user-images.githubusercontent.com/27637970/37023565-66ad9d84-2137-11e8-81d5-3acb16ddddf5.png)

Navigation Menu
![image](https://user-images.githubusercontent.com/27637970/37023595-825eed80-2137-11e8-8fee-6057b799f57f.png)

suada-haji commented on app/src/main/res/layout/nav_header_navigation_drawer.xml:
```diff
@@ -0,0 +1,41 @@
+
+    <de.hdodenhof.circleimageview.CircleImageView
+        android:id="@+id/displayPicture"
+        android:layout_width="80dp"
```
Place values in the `dimens` resource file

Ewanjiru replied:
well noted.

suada-haji commented on app/src/main/res/layout/app_bar_navigation_drawer.xml:
```diff
@@ -0,0 +1,30 @@
+        android:theme="@style/AppTheme.AppBarOverlay">
+
+        <android.support.v7.widget.Toolbar
+            android:id="@+id/toolbarTB"
```
Refrain from using `Caps` in ids

Ewanjiru replied:
Well noted.

suada-haji commented on app/src/main/java/com/andela/art/serialentry/SerialEntryActivity.java:
```diff
@@ -21,7 +21,7 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_serial_entry);
 
-        Toolbar toolbar = findViewById(R.id.toolbarTB);
+        Toolbar toolbar = findViewById(R.id.toolbar);
```
You could implement Data binding here, you can reach to @asce1062  to show you how they did it

Ewanjiru replied:
well noted.

muhallan committed:
feat(navigation menu): add drawables for menu icons

- add drawables in various screen densities
- add menu header background drawable

muhallan committed:
feat(navigation view): add the navigation view

- add the hamburger icon
- add navigation menu items
- display a user's details in the navigation drawer header
- ensure the menu items are clickable
- write tests for the navigation view
[Finishes #155319173, #155319231, #155183816, #155922706]

muhallan committed:
feat(unwanted files): delete no longer needed files

muhallan committed:
chore(gitignore): unstage files that need to be ignored

- delete the files that have already been pushed to upstream
- add those files to .gitignore

muhallan committed:
feat(refactor): move all code to one file

- move all the navigation view code to the serial entry layout
- delete the no longer used files
[Finishes #155319173, #155319231, #155183816, #155922706]
